### PR TITLE
LPS-173206 [EVP] Create Fragment EVP Data Money

### PR DESCRIPTION
**Pull request by Claudionilton**

### CODE REVIEW

**Comment by Claudionilton** [comment]
ci:test:sf

**Comment by eliasantosf** [comment]
Just starting reviewing :)

**Comment by eliasantosf** [comment]
Just starting reviewing :)

**Review by eliasantosf — APPROVED** [review]

**Review by jairmedeiros — APPROVED** [review]

**Comment by jairmedeiros** [comment]
ci:forward

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	"actions": {
+		"delete": {
+		},
+		"get": {
+		},
+		"permissions": {
+		},
+		"update": {
+		}
+	},
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:13]: Probably not necessary

```suggestion
	
```

**Comment by kevenleone** [comment]
ci:stop

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 2]
```diff
@@ -0,0 +1,247 @@
+{
+	"accountEntryRestricted": false,
+	"accountEntryRestrictedObjectFieldName": "",
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:3]:
```suggestion

```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+		"permissions": {
+		},
+		"update": {
+		}
+	},
+	"active": true,
+	"defaultLanguageId": "en_US",
+	"enableCategorization": true,
+	"enableComments": false,
+	"enableObjectEntryHistory": false,
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:18]:
```suggestion
	
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+	"label": {
+		"en_US": "EVP Data"
+	},
+	"name": "EVPData",
+	"objectActions": [
+	],
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:24]:
```suggestion

```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 5]
```diff
@@ -0,0 +1,247 @@
+{
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:1]: `data.json` is too generic

`evp-data.json` is better

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+				"en_US": "Modified Date"
+			},
+			"name": "modifiedDate",
+			"objectFieldSettings": [
+			],
+			"required": false,
+			"state": false,
+			"system": true,
+			"type": "Date"
+		},
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:152]: This field is auto-generated

```suggestion
		
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+				"en_US": "ID"
+			},
+			"name": "id",
+			"objectFieldSettings": [
+			],
+			"required": false,
+			"state": false,
+			"system": true,
+			"type": "Long"
+		},
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:134]: This field is auto-generated

```suggestion
	
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+				"en_US": "Author"
+			},
+			"name": "creator",
+			"objectFieldSettings": [
+			],
+			"required": false,
+			"state": false,
+			"system": true,
+			"type": "String"
+		},
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:61]: This field is auto-generated

```suggestion
		
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+				"en_US": "Create Date"
+			},
+			"name": "createDate",
+			"objectFieldSettings": [
+			],
+			"required": false,
+			"state": false,
+			"system": true,
+			"type": "Date"
+		},
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:43]: This field is auto-generated

```suggestion
		
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+				"en_US": "Status"
+			},
+			"name": "status",
+			"objectFieldSettings": [
+			],
+			"required": false,
+			"state": false,
+			"system": true,
+			"type": "String"
+		},
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/object-definitions/data.json:206]: This field is auto-generated

```suggestion

```

**Review by kevenleone — CHANGES_REQUESTED** [review]
@Claudionilton and @eliasantosf take a look

Thanks 🗡️

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+const dataNumber = document.querySelector('.grantMoney div h2').innerText;
```
[the remark this answers is not in the document]
Comment (Claudionilton) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js]: Done

**Comment by Claudionilton** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+const newDataNumber = parseFloat(+dataNumber);
```
[the remark this answers is not in the document]
Comment (Claudionilton) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js]: Done

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+document.querySelector('.grantMoney div h2').innerHTML = Liferay.Util.escape(
```
[the remark this answers is not in the document]
Comment (Claudionilton) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js]: Done

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+const grantNumber = Number(
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js:11]: Space

```suggestion

const grantNumber = Number(
```

Comment (kevenleone) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js:11]: the variable name can be: 

```suggestion
const grantMoney = Number(
```

Important: Variables names should have clear name, see the example

https://github.com/ryanmcdermott/clean-code-javascript#use-meaningful-and-pronounceable-variable-names

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+	document.querySelector('.grant-money div h2').innerText
+);
+
+document.querySelector('.grant-money div h2').innerHTML = Liferay.Util.escape(
+	new Intl.NumberFormat('en-GB', {
+		compactDisplay: 'short',
+		notation: 'compact',
+	}).format(grantNumber)
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js:19]:
```suggestion
	}).format(grantMoney)
```

Comment (kevenleone) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-evp/src/main/resources/site-initializer/fragments/group/evp/fragments/evp-data-money/index.js:19]: If you put the variable name clear to others to read, it's not necessary to specify the type of the data.

**Review by kevenleone — CHANGES_REQUESTED** [review]
@Claudionilton take a look

The rest is fine, this is just a minor change.

Take a look into this repository: https://github.com/ryanmcdermott/clean-code-javascript

Explain so many things about clean code, it's very helpful

**Comment by Claudionilton** [comment]
ci:test:sf

**Review by kevenleone — APPROVED** [review]
LGTM 🔥 

Good job @Claudionilton and to others involved 😄 

🎉

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**